lib/teiserver/o_auth/tasks/gen_token.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Teiserver.OAuth.Task.GenToken do
+  @moduledoc """
+  Generates a token for the given user. This is for a convenience task to ease
+  the development of anything requiring OAuth tokens like tachyon protocol.
+  """
+
+  alias Teiserver.OAuth.{ApplicationQueries, Token}
+
+  @spec create_token(String.t(), String.t() | nil) :: {:ok, Token.t()} | {:error, term()}
+  def create_token(username_or_email, app_uid \\ nil) do
+    with {:ok, user} <- get_user(username_or_email),
+         {:ok, app} <- get_app(app_uid) do
+      token_attr = %{
+        value: Base.hex_encode32(:crypto.strong_rand_bytes(32), padding: false),
+        owner_id: user.id,
+        application_id: app.id,
+        scopes: app.scopes,
+        expires_at: Timex.add(DateTime.utc_now(), Timex.Duration.from_days(1)),
+        type: :access,
+        refresh_token: nil
+      }
+
+      %Token{}
+      |> Token.changeset(token_attr)
+      |> Teiserver.Repo.insert()
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  defp get_user(username_or_email) do
+    user =
+      if String.contains?(username_or_email, "@") do
```

Comment:
Why not use `UserCacheLib.get_user_by_name` and `UserCacheLib.get_user_by_email`?

Reply:
Originally I had issues because the caches were not setup. But it seems that it's fine now, not sure how though.
Anyway, I replaced the queries with the higher level ones using cache.